### PR TITLE
Made tasks

### PR DESCRIPTION
**Pull request by Max-Gosling**

### CODE REVIEW

File: task_05/src/topology_sort.hpp [hunk 1]
```diff
@@ -1 +1 @@
-#pragma once
```
Comment (LostPointer) [inline task_05/src/topology_sort.hpp:1]: 🔴 **Откатить это изменение**

Файл был `#pragma once`, после PR стал пустым. `task_05/src/test.cpp` делает `#include "topology_sort.hpp"` — это сломает сборку при будущих изменениях task_05. Task_05 не входит в скоуп этого PR, файл трогать не нужно.

File: task_01/src/search.hpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#include <utility>
```
Comment (LostPointer) [inline task_01/src/search.hpp:1]: 🔴 **Отсутствует `#pragma once`** — и в этом файле, и во всех остальных `.hpp` этого PR:
`stack.hpp`, `finder.hpp` (task_02), `phone.hpp`, `sort.hpp`, `finder.hpp` (task_11), `coin.hpp`

Без header guard при двойном включении компилятор получит дублирующиеся определения → ошибка компиляции. Каждый `.hpp` должен начинаться с `#pragma once`.

File: task_01/src/search.hpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+
+
+auto FindTwoSum(int expected, int size, int* nums)
```
Comment (LostPointer) [inline task_01/src/search.hpp:5]: 🟠 **Реализация функции в `.hpp`**

Функция с телом в заголовке — нарушение ODR при включении в несколько единиц трансляции. Стандарт проекта: объявление в `.h`, реализация в `.cpp`. Пример правильной структуры — `task_07/sort.hpp` + `sort.cpp`.

File: task_01/src/stack.hpp [hunk 4]
```diff
@@ -0,0 +1,17 @@
+#include <vector>
```
Comment (LostPointer) [inline task_01/src/stack.hpp:1]: 🟠 **Лишний файл**

`stack.hpp` нигде не используется — ни в тестах, ни в `main.cpp`. Удалить.

File: task_02/src/test.cpp [hunk 5]
```diff
@@ -1,5 +1,16 @@
 #include <gtest/gtest.h>
+#include "finder.hpp"
 
-TEST(Test, Simple) {
-  ASSERT_EQ(1, 1);  // placeholder
+TEST(FindBounder, Simple) {
```
Comment (LostPointer) [inline task_02/src/test.cpp:4]: 🟠 **Минимальное покрытие: только 2 теста**

Отсутствуют edge cases:
- Массив из одних нулей → граница на последнем индексе
- Массив из одних единиц → некорректный ввод, поведение не определено
- Один элемент

File: task_07/src/main.cpp [hunk 6]
```diff
@@ -1,3 +1,19 @@
...
+int main() {
+    int N;
+    std::cin >> N;
+    std::vector<int> nums(N);
+    for (int i = 0; i < N; ++i) {
+        std::cin >> nums[i];
+    }
+    QuickSort(N, nums);
+    for (auto i = 0; i < nums.size(); ++i) {
+        if (i > 0) std::cout << ' ';
```
Comment (LostPointer) [inline task_07/src/main.cpp:14]: 🟡 **Signed/unsigned comparison**

`auto i = 0` выводится как `int`, `nums.size()` возвращает `size_t` (unsigned) → предупреждение компилятора. Исправить:
```cpp
for (int i = 0; i < (int)nums.size(); ++i)
```

**Review by LostPointer — CHANGES_REQUESTED** [review]
## Обзор PR

Алгоритмы в целом корректны, тесты покрывают основные случаи. Но есть системные проблемы со структурой файлов, которые нужно исправить перед мержем.

**Хорошее:**
- `task_07`: единственная задача с правильной структурой `.cpp` + `.h` — так должно быть во всех задачах
- `task_12` MinCoins — корректная DP-реализация
- `task_11`: исправлен `basic.out` (4 → 5)
- `task_02` FindIdOfBounder: бинарный поиск с корректным `mid = left + (right-left+1)/2`
- `task_11`: 6 тестов с разными сценариями

**Форматирование:** весь PR использует 4-пробельные отступы — нужно прогнать `./format.sh` (Google style = 2 пробела). Большинство файлов без trailing newline.
